configs/solc-bin-ci-reference.json

```diff
@@ -0,0 +1,16 @@
+{
+  "binaries": {
+    "${SOLC_VERSION}": {
+      "is_enabled": true,
+      "protocol": "https",
+      "source": "https://github.com/matter-labs/era-solidity/releases/download/${VERSION}/solc-${PLATFORM}-${VERSION}-${ZKSYNC_REVISION}",
```

Comment:
So are these `VERSION` or `SOLC_VERSION`?

Reply:
It's a base version of `solc` without `ZKSYNC_REVISION`.

Reply:
I guess we need only 2 variables: `VERSION` and `ZKSYNC_REVISION`.

Reply:
I'll change `${SOLC_VERSION}` in CI to e.g. `0.8.26` and then `era-compiler-tester` will expand `${VERSION}` to the right value.

Reply:
What about this? `${VERSION}-${ZKSYNC_REVISION}`

Reply:
But does `era-compiler-tester` know it? What should it expand it with?

Reply:
That's why initially I didn't want to use `${}` here to mix things up. Right now it's impossible to understand what is supported by tester and what is not.

Reply:
The thing is not about `${}`.
It's just the tester doesn't know the revision unless I'll be hardcoding it there.

Reply:
For now tester only cares about `${PLATFORM}`.

Reply:
The idea of this config is to provide a way to hardcode it through `sed` in CI.

Reply:
So let's do this.
I'm only asking about what each variable is supposed to be substituted with.

Reply:
Maybe, we can do better somehow... I don't know.

Reply:
Could you just give an example of substitution?
It is very important where each value is taken from.

Reply:
In fact, I doubt we should be downloading the reference from anywhere.
Would it be possible to build the base branch, just like we're doing in the tester and LLVM?

Reply:
I was thinking of taking `${SOLC_VERSION}` from the `base_ref` (base branch of the PR), and for the reference version of `${ZKSYNC_REVISION}` introduce a variable in CI with const value of `1.0.0` for now that we'll have to update in one place each time we update the version.

Reply:
I'm actually thinking right now that probably it's easier to update `"is_enabled"` to `true` for a particular version.

Reply:
I agree regarding, `${ZKSYNC_REVISION}`, but:
1. This will fail if there are two levels of branches.
2. `reference` is supposed to be the base branch, but we're downloading the release. (it is my fault and legacy, but we now may clean this up)

`${SOLC_VERSION}` can also be hardcoded in the CI I guess.

Reply:
> "reference is supposed to be the base branch"

This is actually solves the problem completely. If I build the base branch version for the `reference`, then we don't need any custom config and filter will work.

Reply:
But does it solve #1? Base branch doesn't always have to be a version branch.

Reply:
No, #1 will stay unsolved. We can solve it by taking the version from the CMake project as the `solc` itself does:
`set(PROJECT_VERSION "0.4.12")`

Reply:
Then, it will work for any branch.

Reply:
@antonbaliasnikov can we lift it from cmake and hardcode it next to ZKSYNC_REVISION?

Reply:
Awesome, let's take it from `cmake` then!

Reply:
I guess it is always set by the upstream developers, so we don't have to care about updating it.

Reply:
Let's close this PR then.